File: .guides/tests/ch-2.js

keyPressedEvent  = null;

$.getScript(window.location.origin + '/public/js/' + window.testEnv.cmd + '.js?_=' + Date.now())
.done(function (script, status) {
  
  createEmptyMaze();
    
  var _left = false;
  var _right = false;
  var _up = false;
  var _down = false;
  var _another = false;
  
  if(typeof keyPressedEvent  == 'function') {
    
    window.showMessage = function (val) {
      if(val.toLowerCase() == 'horizontal') _left = true;
    }

    keyPressedEvent('LEFT');


    window.showMessage = function (val) {
      if(val.toLowerCase() == 'horizontal') _right = true;
    }

    keyPressedEvent('RIGHT');
    

    window.showMessage = function (val) {
      if(val.toLowerCase() == 'vertical') _up = true;
    }

    keyPressedEvent('UP');

    
    window.showMessage = function (val) {
      if(val.toLowerCase() == 'vertical') _down = true;
    }

    keyPressedEvent('DOWN');


    window.showMessage = function (val) {
      if(val.toLowerCase() == 'another key was pressed') _another = true;
    }

    keyPressedEvent('C');

    
    if(!_left || !_right || !_up || !_down || !_another) {
      return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }
  }
  else {
    return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
  }
  
  
  codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.SUCCESS, 'Well done!');    

})
.fail(function (jqxhr, settings, exception) {
  console.log(exception);
  codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.INVALID, exception.message); 
});

File: .guides/tests/ch-2.py

energy = 9
steps = 10
message = ''

def showMessage(val):
  global message
  message = val

try:
  execfile('/home/codio/workspace/public/py/ch-2.py')
  
  keyPressedEvent('UP')
  if message.lower().strip() != 'vertical':
    raise ValueError("keyPressedEvent is not handling the 'UP' keyCode correctly")

  keyPressedEvent('DOWN')
  if message.lower().strip() != 'vertical':
    raise ValueError("keyPressedEvent is not handling the 'DOWN' keyCode correctly")

  keyPressedEvent('LEFT')
  if message.lower().strip() != 'horizontal':
    raise ValueError("keyPressedEvent is not handling the 'LEFT' keyCode correctly")

  keyPressedEvent('RIGHT')
  if message.lower().strip() != 'horizontal':
    raise ValueError("keyPressedEvent is not handling the 'RIGHT' keyCode correctly")

  keyPressedEvent('OTHER')
  if message.lower().strip() != 'another key was pressed':
    raise ValueError("keyPressedEvent is not handling other keyCodes correctly")
    
  print 'well done'
  exit(0)
  
except (IOError, SyntaxError, ValueError) as e:
  print str(e)
  exit(1)
  
print 'Not quite right, try again!'
exit(1)
